### PR TITLE
Edits to Meat and Conclusion

### PR DESCRIPTION
**Pull request by willfanguy**
Ran edits for the-meat, the-mean-start, and conclusion.

### CODE REVIEW

File: manuscript/conclusion.txt [hunk 1]
```diff
@@ -2,12 +2,14 @@
 
 If you've followed the examples, you've just created your first visualization with React and d3.js. You’re amazing!
 
-If you've only read through the book, that’s awesome too. You know much more about making reusable visualizations than you did an hour ago.
+If you've only read through the book, that’s awesome, too. You know much more about making reusable visualizations than you did an hour ago.
 
 You understand how the declarative nature of React works really well with d3.js and vice-versa. You know what it takes to write reusable components and how much easier they make your life once you’ve built them. And you know how to configure grunt, browserify, and everything else to create an awesome environment to work with.
 
-In short, you know everything you need to start making your own visualization components. At the very least, you know whether React and d3.js good tech stack for you.
+ {>>You'll still need to change the references to `grunt` and `browserify` up there.<<}
```
Comment (Swizec) [inline manuscript/conclusion.txt:9]: Please feel free to make changes like that if you see them.

File: manuscript/the-meat-start.txt [hunk 2]
```diff
@@ -38,17 +38,15 @@ Which compiles to:
         document.querySelectorAll(‘.h1bgraph’)[0]
     );
 
-As you can see, HTML code translates to `React.createElement` calls with attributes translated into a property dictionary. The beauty of this approach is that you can use React components as if they were HTML tags and that HTML attributes can be anything.
+As you can see, HTML code translates to `React.createElement` calls with attributes translated into a property dictionary. The beauty of this approach is two-pronged: you can use React components as if they were HTML tags and  HTML attributes can be anything.
```
Comment (Swizec) [inline manuscript/the-meat-start.txt]: `tags and  HTML attributes`

You created a double-space. I'm not sure if that matters or not when this gets compiled into a pdf.

File: manuscript/the-meat-start.txt [hunk 3]
```diff
@@ -38,17 +38,15 @@ Which compiles to:
...
+I'm not sure yet whether this is better than separate template files in Mustache or whatever. There are benefits to both approaches. I mean, would you let a designer write the HTML inside your JavaScript files? I wouldn't, but it's definitely better than manually `+`-ing strings or Angular's approach of putting everything into HTML. Considerably better.
 
-But it's definitely better than manually `+`-ing strings or Angular's approach of putting everything into HTML. Much better.
+If you skipped the setup section and don't have a JSX compilation system set up, you should do that now. You can also use the project stub you got with the book.
 
-If you skipped the setup section and don't have a JSX compilation system set up, I suggest you do that now. You can also use the project stub you got with the book.
-
-Just like the official documentation, I don't suggest using `JSXTransform.js`. Sure it's neat to have all this compilation happen right in the browser, but it's bad practice.
+Just like the official documentation, I don't suggest using `JSXTransform.js`. Sure, it's neat to have all this compilation happen right in the browser, but it's bad practice.
```
Comment (Swizec) [inline manuscript/the-meat-start.txt]: We should remove this whole paragraph. It isn't all that relevant, doesn't add much, and is potentially confusing.

File: manuscript/the-meat-start.txt [hunk 4]
```diff
@@ -75,9 +73,9 @@ If the parent component puts this inside an `<svg>` element, the user will see a
 
 The d3.js approach outputs SVG as if by magic. It also looks a bit cleaner because it's pure JavaScript. But it’s a lot more typing and function calls for the same result.
```
Comment (Swizec) [inline manuscript/the-meat-start.txt:72]: I have to improve this paragraph. It doesn't flow well.

File: manuscript/the-meat-start.txt [hunk 5]
```diff
@@ -75,9 +73,9 @@ If the parent component puts this inside an `<svg>` element, the user will see a
...
 
-Well, actually, the pure d3.js example is 10 characters shorter. But trust me, React is cool, okay.
+Well, actually, the pure d3.js example is 10 characters shorter. But trust me, React is way cooler.
 
-Anyway, my point is that dealing with the DOM is not d3.js’s strong suit. Especially once you're drawing a few thousand elements and your visualization slows down to a leisurely stroll if you’re careful.
+My point is that dealing with the DOM is not d3.js’s strong suit, especially once you're drawing a few thousand elements and your visualization slows down to a leisurely stroll if you’re careful.
```
Comment (Swizec) [inline manuscript/the-meat-start.txt]: Would this flow better with an ellipsis before `if you're careful`?

File: manuscript/the-meat-start.txt [hunk 6]
```diff
@@ -86,14 +84,14 @@ We're going to follow this simple approach:
 * React owns the DOM
 * d3 calculates properties
 
-This way we leverage both React and d3.js for what they’re best at.
+This way we leverage both React and d3.js for their best functions.
```
Comment (Swizec) [inline manuscript/the-meat-start.txt]: I think this will be better without "for their best functions". Leveraging already implies bestness and appropriateness of use.

File: manuscript/the-meat-start.txt [hunk 7]
```diff
@@ -86,14 +84,14 @@ We're going to follow this simple approach:
...
 
 {#the-architecture}
 ## The architecture
 
-To make our lives easier we're going to use a flow-down architecture where the entire application state is stored in one place. It's inspired by Flux, but uses less code and is easier to explain. The downside is that it doesn't scale as well as Flux would.
+To make our lives easier, we're going to use a flow-down architecture where the entire application state is stored in one place. It's inspired by Flux, but it uses less code and is easier to explain. The downside is that it doesn't scale as well as Flux would.
```
Comment (Swizec) [inline manuscript/the-meat-start.txt]: I use `it` too often here. Suggestions?

File: manuscript/the-meat-start.txt [hunk 8]
```diff
@@ -105,30 +103,30 @@ The idea is this:
...
 
-If you want to read about this in more detail Google for "isomorphic JavaScript". A good buzzword is also "referential transparency" and "idempotent functions".
+If you want to read about this in more detail, Google "isomorphic JavaScript". You could also search for "referential transparency" and "idempotent functions".
 
 Either way, functional programming for HTML. Yay!
 
 ## The HTML skeleton
 
-We’re building the whole interface with React, but we still need some HTML to start with. It's going to take care of including files and giving our UI a container.
+We’re building the whole interface with React, but we still need to being with some HTML. It's going to take care of including files and giving our UI a container.
```
Comment (Swizec) [inline manuscript/the-meat-start.txt]: You probably meant `begin` :)

File: manuscript/the-meat.txt [hunk 9]
```diff
@@ -1,21 +1,21 @@
...
 We start by adding three short methods to `H1BGraph`:
 
 {crop-start-line=34,crop-end-line=49,linenos=off,lang=jsx}
 <<[Base methods to load data](code_samples/main.jsx)
 
-React calls `componentWillMount` right before inserting the component into our document. It’s a great opportunity to make any last minute preparations before rendering. In our case, it’s a good spot to start loading data. 
+React calls `componentWillMount` right before inserting the component into our document. It’s a great opportunity to make any last-minute preparations before rendering. In our case, it’s a good spot to start loading data.
 
-I'm sure better spots exist, but loading a 12MB data file is always going to take a couple of seconds. And you should avoid loading data you never intend to render.
+I'm sure better spots exist, but loading a 12MB data file is always going to take a couple of seconds, and you should avoid loading data you never intend to render.
```
Comment (Swizec) [inline manuscript/the-meat.txt]: This sentence feels too long. Ideas for what we can cut?

File: manuscript/the-meat.txt [hunk 10]
```diff
@@ -26,7 +26,7 @@ This will asynchronously load a CSV file, parse it, and return the result in the
 
 Using `setState()` triggers a re-render and should generally be avoided because relying on state can cause inconsistencies in your UI. It’s very appropriate to store 12MB of raw data as state, though.
 
-But the data we’ve got now has keys with spaces like `submit date` and `job title`. And everything is a string. Not very nice to work with.
+So the data we’ve got now has keys with spaces like `submit date` and `job title`, and everything is a string. This is less than ideal, and not very nice to work with.
```
Comment (Swizec) [inline manuscript/the-meat.txt]: We can drop the connecting first word since it doesn't really add anything. I'd also change last sentence to `This is not very nice to work with`.

File: manuscript/the-meat.txt [hunk 11]
```diff
@@ -66,14 +66,14 @@ Into objects that look like this:
 			    "state": "ca"
 			}
 
-In our case we cleaned up the keys, parsed dates into `Date()` objects and made sure numbers are numbers. If a row didn’t have a `base_salary`, we filtered it out by returning `null`.
+In our case we cleaned up the keys, parsed dates into `Date()` objects, and made sure numbers are numbers. If a row didn’t have a `base_salary`, we filtered it out by returning `null`.
 
-But loading and parsing 81,000 datapoints takes some time. Let's tell users what they're waiting for by rendering some explainer text when there's no data.
+But loading and parsing 81,000 data points takes some time. Let's tell users what they're waiting for by rendering some explainer text when there's no data.
```
Comment (Swizec) [inline manuscript/the-meat.txt]: Drop the first `But`.

File: manuscript/the-meat.txt [hunk 12]
```diff
@@ -94,39 +94,39 @@ We start with a blank component called `Histogram`.
...
 We add three methods to our `Histogram` component to manage d3.js:
 
 {crop-start-line=26,crop-end-line=45,linenos=off,lang=jsx}
 <<[D3.js management functions](code_samples/drawers.jsx)
 
-This is the cleanest approach I’ve found to make d3.js and React work together. We’ll use it for all our d3.js-enabled components.
-	
+This is the cleanest approach I’ve found to make d3.js and React work together. We’ll use it for all of our d3.js-enabled components.
+
 The problem we're facing is that the way d3.js achieves its declarative nature is through objects. When you call a d3.js function, it's usually going to have some internal state.
```
Comment (Swizec) [inline manuscript/the-meat.txt:112]: We can clean up the first sentence as `The problem we're facing is that d3.js achieves its declarative nature through objects`.

File: manuscript/the-meat.txt [hunk 13]
```diff
@@ -94,39 +94,39 @@ We start with a blank component called `Histogram`.
...
 
-Which is great when you want to change how it behaves by daisy chaining getters and setters. But it causes problems with React because it means we have to make sure the internal state is always up to date.
+This is great when you want to change how it behaves by daisy-chaining getters and setters, but it causes problems with React because it means we have to make sure the internal state is always up-to-date.
```
Comment (Swizec) [inline manuscript/the-meat.txt]: Second part of this sentence is a mouthfull. Ideas?

File: manuscript/the-meat.txt [hunk 14]
```diff
@@ -94,39 +94,39 @@ We start with a blank component called `Histogram`.
...
 
-We do this by relying on `componentWillMount`, to set the defaults, and then `componentWillRecieveProps`, to update them when necessary.
+We do this by relying on `componentWillMount` to set the defaults, and then `componentWillRecieveProps` to update them when necessary.
```
Comment (Swizec) [inline manuscript/the-meat.txt]: Let's change the first part to `We'll use componentWillMount to set the defaults`

File: manuscript/the-meat.txt [hunk 15]
```diff
@@ -188,20 +188,20 @@ You should now see a histogram like this:
 
 ## Adding an axis
 
-Axes are one of my favourite feats of d3.js magic. You just call a function, set some parameters and bam, you’ve got an axis.
+Axes are one of my favourite feats of d3.js magic. You simply call a function, set some parameters, and **BAM**, you’ve got an axis.
```
Comment (Swizec) [inline manuscript/the-meat.txt]: Let's drop the `simply` it doesn't add anything.

In general I think most `just`s and `simply`s and things like that can be removed. Please call me out on my shit and tell me when I'm using words that don't add anything to the sentence.

File: manuscript/the-meat.txt [hunk 16]
```diff
@@ -210,25 +210,25 @@ We created a scale and an axis. The axis will use our linear scale, will be orie
...
+Finally, here comes the dirty trick:
 
 {crop-start-line=369,crop-end-line=381,linenos=off,lang=jsx}
 <<[The dirty trick](code_samples/drawers.jsx)
 
-I’m sure this goes against everything React designers fought for, but it works. We hook into the `componentDidUpdate` and `componentDidMount` callbacks with a `renderAxis` method. Then we use `this.getDOMNode()` to get a reference to the rendered element, feed it into `d3.select()` and `.call()` the axis on it.
+I’m sure this goes against everything React designers fought for, but it works. We hook into the `componentDidUpdate` and `componentDidMount` callbacks with a `renderAxis` method. Then, we use `this.getDOMNode()` to get a reference to the rendered element, feed it into `d3.select()`, and `.call()` the axis on it.
 
-What we’re doing is basically throwing away the whole element and creating it from scratch on every render. Rather inefficient, but it’s okay if used sparingly.
+What we’re doing is basically throwing away the whole element and creating it from scratch on every render. It's rather inefficient, but it’s okay if it's used sparingly.
```
Comment (Swizec) [inline manuscript/the-meat.txt]: Would the last part work better if we replace `if it's` with `when`?

File: manuscript/the-meat.txt [hunk 17]
```diff
@@ -283,34 +283,34 @@ Let’s start a new file called `src/controls.jsx` with a `Controls` component s
...
-This will break in your browser because we don’t have a `ControlRow` component yet. The idea is to have a component that takes data, a function that generates the button labels and an `updateDataFilter` callback.
+This will break in your browser because we don’t have a `ControlRow` component yet. The idea is to have a component that takes data, a function that generates the button labels, and an `updateDataFilter` callback.
 
 We’ll define the callback later. Let’s make the `ControlRow` component first.
 
 {crop-start-line=51,crop-end-line=72,linenos=off,lang=jsx}
 <<[ControlRow component](code_samples/controls.jsx)
 
-Nothing special going on here. We’ve got a `render` method that returns a Bootstrap row and fills it with with a bunch of `Toggle` components. It does so by looping over the result of `this.props.getToggleNames`, which is the function we defined in `Controls.render`.
+There's nothing special going on here. We’ve got a `render` method that returns a Bootstrap row and fills it with with a bunch of `Toggle` components. It does so by looping over the result of `this.props.getToggleNames`, which is the function we defined in `Controls.render`.
```
Comment (Swizec) [inline manuscript/the-meat.txt]: Double `with with`

File: manuscript/the-meat.txt [hunk 18]
```diff
@@ -366,27 +366,27 @@ Nothing fancy, just a call to `this.props.updateDataFilter` with information on
...
 
-But we’ve just created an infinite loop. When `componentDidUpdate` is called we call something on the parent component. This potentially changes our `props`, which again triggers `componentDidUpdate`.
+Ok, we’ve just created an infinite loop. When `componentDidUpdate` is called we call something on the parent component. This potentially changes our `props`, which again triggers `componentDidUpdate`.
 
 We have to make sure the component updates *only* when properties have actually changed. The easiest way to do that is using `shouldComponentUpdate`. If it returns `false`, the component doesn’t update.
 
-An apparently better option is using the [PureRenderMixin](http://facebook.github.io/react/docs/pure-render-mixin.html), which makes sure components only update when properties actually change. Could be worth a shot.
+A possibly better option is using the [PureRenderMixin](http://facebook.github.io/react/docs/pure-render-mixin.html), which makes sure components only update when properties actually change. It could be worth a shot.
```
Comment (Swizec) [inline manuscript/the-meat.txt]: This paragraph should be removed. Mixins aren't popular anymore.

File: manuscript/the-meat.txt [hunk 19]
```diff
@@ -400,15 +400,15 @@ The last step is filtering the data in our `H1BGraph.render` method.
 {crop-start-line=530,crop-end-line=552,linenos=off,lang=jsx}
 <<[Filter data in H1BGraph.render](code_samples/main.jsx)
 
-As you can see, we didn’t have to do much. We just passed the `rawData` through our filter before feeding it to the `Histogram.
+As you can see, we didn’t have to do much. We just passed the `rawData` through our filter before feeding it to the `Histogram`.
```
Comment (Swizec) [inline manuscript/the-meat.txt]: the `just` can be removed, no?

File: manuscript/the-meat.txt [hunk 20]
```diff
@@ -434,7 +434,7 @@ Let’s add them.
 
 There’s nothing special going on here. `getYears` uses `_.groupBy` to group the data by year, then returns the keys of the array. We could make this more efficient, but it works well enough and is easy to read.
 
-`getFormatter` relies on the fact that d3.js scales have formatters. It creates a linear scale with a domain, and returns the `tickFormat()` function. Formatters don't work well without a defined domain.
+`getFormatter` relies on the fact that d3.js scales have formatters. It creates a linear scale with a domain and returns the `tickFormat()` function. Formatters don't work well without a defined domain.
```
Comment (Swizec) [inline manuscript/the-meat.txt:435]: We can get rid of `the fact that` by changing `have` into `having`. I think that improves the flow. Does it?

File: manuscript/the-meat.txt [hunk 21]
```diff
@@ -461,14 +461,14 @@ We start with the year fragment:
...
-The render function is going to be simple as well:
+The `render` function is going to be simple as well:
 
 {crop-start-line=110,crop-end-line=129,linenos=off,lang=jsx}
 <<[Title.render function](code_samples/meta.jsx)
 
 We’re returning an `<h2>` element with a title that includes some dynamic tags. We change `made` to `make` based on whether or not we’re adding `in years`, and we used `d3.mean` to get the average.
 
-Now we can just add the title into H1BGraph’s `render` method. You can use as many as you want, wherever you want, but I think putting it above the graph works best.
+Now we can just add the title into H1BGraph’s render method. You can use as many as you want, wherever you want, but I think putting it above the graph works best.
```
Comment (Swizec) [inline manuscript/the-meat.txt]: The `just` is just filler here, right?

File: manuscript/the-meat.txt [hunk 22]
```diff
@@ -483,14 +483,14 @@ Now let’s add a paragraph of description to make the story clearer.
 
 ### The description
 
-Our descriptions won’t be much more complicated than the titles. Just a sentence or two explaining what the histogram is showing and comparing it to the previous year.
+Our descriptions won’t be much more complicated than the titles. We'll just need a sentence or two explaining what the histogram is showing and comparing it to the previous year.
```
Comment (Swizec) [inline manuscript/the-meat.txt]: Same question as above with this `just`.

File: manuscript/the-meat.txt [hunk 23]
```diff
@@ -513,21 +513,21 @@ But the `getPreviousYearFragment` function is more complex:
 
         return fragment;
     },
+**
```
Comment (Swizec) [inline manuscript/the-meat.txt]: Why is this `**` here?

File: manuscript/the-meat.txt [hunk 24]
```diff
@@ -513,21 +513,21 @@ But the `getPreviousYearFragment` function is more complex:
...
+First, we get the list of years. If it’s empty or the current year is 2012 (the first in our dataset), we return an empty fragment.
 
-First we get the list of years. If it’s empty or the current year is 2012 (the first in our dataset), we return an empty fragment.
-
-Otherwise we get last year’s data with `this.getAllDataByYear()`, then compare the number of entries with this year. We return a fragment that’s either “X% **more** than the year before” or “X% **less** than the year before”.
+Otherwise, we get last year’s data with `this.getAllDataByYear()`, then compare the number of entries with this year. We return a fragment that’s either “X% **more** than the year before” or “X% **less** than the year before”.
 
 Now we’ve just got to use these two fragments in the render method.
```
Comment (Swizec) [inline manuscript/the-meat.txt:519]: I don't like this sentence. Ideas?

File: manuscript/the-meat-start.txt (hunk 2 above)
Comment (willfanguy) [inline manuscript/the-meat-start.txt]: Fixed.

File: manuscript/the-meat-start.txt (hunk 3 above)
Comment (willfanguy) [inline manuscript/the-meat-start.txt]: Done.

File: manuscript/the-meat-start.txt (hunk 5 above)
Comment (willfanguy) [inline manuscript/the-meat-start.txt]: Done.

File: manuscript/the-meat-start.txt (hunk 6 above)
Comment (willfanguy) [inline manuscript/the-meat-start.txt]: "This way we leverage both React and d3.js for their best functions and not much else."

-What about that?

File: manuscript/the-meat-start.txt (hunk 7 above)
Comment (willfanguy) [inline manuscript/the-meat-start.txt]: > > "To make our lives easier, we're going to use a flow-down architecture where the entire application state is stored in one place. The architecture itself inspired by Flux, but the way we're structuring it uses less code and is easier to explain. The downside is that our version doesn't scale as well as Flux would."

File: manuscript/the-meat-start.txt (hunk 8 above)
Comment (willfanguy) [inline manuscript/the-meat-start.txt]: Fixed it. Oops. ¯_(ツ)_/¯

File: manuscript/conclusion.txt (hunk 1 above)
Comment (willfanguy) [inline manuscript/conclusion.txt:9]: > > You understand how the declarative nature of React works really well with d3.js and vice-versa. You know what it takes to write reusable components and how much easier they make your life once you’ve built them. And you know how to configure Webpack, Babel, and everything else to create an awesome environment to work with.

File: manuscript/the-meat.txt (hunk 9 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: > > I'm sure better spots exist, but loading a 12MB data file is always going to take a couple of seconds. You should try to avoid loading data you never intend to render.

File: manuscript/the-meat.txt (hunk 10 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: Done

File: manuscript/the-meat.txt (hunk 11 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: Got it.

File: manuscript/the-meat.txt (hunk 12 above)
Comment (willfanguy) [inline manuscript/the-meat.txt:112]: So take out line 110 altogether?

File: manuscript/the-meat.txt (hunk 13 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: > > This is great when you want to change how d3.js behaves by daisy-chaining getters and setters, but that approach causes problems with React. It means we have to make sure the internal state is always up-to-date.

File: manuscript/the-meat.txt (hunk 14 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: Got it.

File: manuscript/the-meat.txt (hunk 15 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: Fixed that first part.

That second part: noted. I was treading lightly when it came to the book. :)

File: manuscript/the-meat.txt (hunk 16 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: It would, so I changed it. :smile:

File: manuscript/the-meat.txt (hunk 17 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: :+1:

File: manuscript/the-meat.txt (hunk 18 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: :+1:

File: manuscript/the-meat.txt (hunk 19 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: :+1:

File: manuscript/the-meat.txt (hunk 20 above)
Comment (willfanguy) [inline manuscript/the-meat.txt:435]: I think it complicates things. `that` separates the two parts into separate clauses. For once, I left it how it is.

File: manuscript/the-meat.txt (hunk 21 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: Yep. I'm going to run the text through Marked 2 in the future with a highlight for `just` and `simply` and the like. That'll help me/us see how often they're being used and will help eliminate the, and this is your word, "shit". :smile:

File: manuscript/the-meat.txt (hunk 22 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: :+1:

File: manuscript/the-meat.txt (hunk 23 above)
Comment (willfanguy) [inline manuscript/the-meat.txt]: Shit. That was a "fix" on my end to prevent the in-line code from fucking with my MD syntax highlighting. I  thought I went through and removed them. Sorry.

File: manuscript/the-meat.txt (hunk 24 above)
Comment (willfanguy) [inline manuscript/the-meat.txt:519]: > > First, we get the list of years. If it’s empty or the current year is 2012 (the first in our dataset), we return an empty fragment.
> > 
> > In any other case, we get a look at last year’s data with `this.getAllDataByYear()`. We can then compare the number of entries with this year and return a fragment that’s either “X% **more** than the year before” or “X% **less** than the year before”.

**Comment by willfanguy** [comment]
Issuing another pull request with the changes.

File: manuscript/the-meat.txt (hunk 20 above)
Comment (Swizec) [inline manuscript/the-meat.txt:435]: :+1:
